src/DonationForms/resources/app/fields/SectionNode.tsx

```diff
@@ -19,10 +19,14 @@ const formTemplates = window.givewp.form.templates;
  */
 function SectionNode({node}: {node: Node}) {
     const showNode = useVisibilityCondition(node.visibilityConditions);
-    const {unregister} = window.givewp.form.hooks.useFormContext();
+    const {unregister, setValue} = window.givewp.form.hooks.useFormContext();
 
     useEffect(() => {
         if (showNode) {
+            if (isField(node) && node.defaultValue !== undefined) {
+                setValue(node.name, node.defaultValue);
+            }
+
```

Comment:
@JasonTheAdams I'm going to suggest a different strategy that should simplify this:

```
unregister(node.name, {keepDefaultValue: true});
```

When the form first renders, we are handing over the default values for every field - regardless of their visibility conditions - so they already exist in the eyes of RHF.  

When a field then gets unregistered (even initially) it appears RHF is removing it's default value from that initial state.  Using the [unregister](https://react-hook-form.com/docs/useform/unregister) option `keepDefaultValue` we're effectively telling RHF to preserve that default value of that field - and use it during registration (when the field is rendered).

Reply:
[Confirmed](https://github.com/react-hook-form/react-hook-form/blob/5f3df1227c381a9127ab27f7d1d98a9cefee33d5/src/logic/createFormControl.ts#L908), when calling unregister - RHF removes the default value from context using the field name.

When calling register - it [pulls](https://github.com/react-hook-form/react-hook-form/blob/5f3df1227c381a9127ab27f7d1d98a9cefee33d5/src/logic/createFormControl.ts#L257C13-L257C25) the default value from context.

Reply:
And in swoops @jonwaldstein, taking this PR down to 3 lines. 😆